k8s/jiva/snapshot-cleanup.sh

```diff
@@ -0,0 +1,88 @@
+        echo "Do you want to continue (Y/N)"
+        read access
+        if [ $access == "N" ]; then
+        exit 1;
+        fi
+    }
+    delete_jiva_snapshot()
+    {   
+        warning
+        min_required_snapshot=4
+        count=0
+        snapshot_number_cmd='kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls | grep -v ID | wc -l'
+        snapshot_name_cmd='kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls | grep -v ID | tail -1'
+
+        if [ $(eval $snapshot_number_cmd) -le $((number_of_snapshot-1)) ]; then
+            echo "Provided number of snapshots are not present"
+            exit 1 ;
+        else
+                    if [ $(eval $snapshot_number_cmd) -gt $(($min_required_snapshot + $number_of_snapshot)) ]; then
+                        block_service
+                        while [ $count -lt $number_of_snapshot ]
+                        do 
+                            snapshot_name=$(eval $snapshot_name_cmd | tr -d '\r')
+                            del_cmd="kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot rm $snapshot_name"
+                            eval $del_cmd
+                            count=$((count+1))
+                            validate_snap_delete $snapshot_name
+                        done
+                        unblock_service
+                    else
+                        echo "Greater number of snapshot is requested for deletion. Try with less number of snapshot" ;
+                    fi
+            fi
+    }
+
+    block_service()
+    {
+        kubectl patch svc $ctrl_service_name -n $pvc_namespace --type merge -p "$(cat tmp.json)"  
+    }
+
+    unblock_service()
+    {
+        sed -i 's/50000/3260/g' tmp.json
+        kubectl patch svc $ctrl_service_name -n $pvc_namespace --type merge -p "$(cat tmp.json)"
+        rm tmp.json
+    }
+
+    validate_snap_delete()
+    {
+        del_snapshot_name=$1
+        validate_cmd='kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls | grep $del_snapshot_name'
+        eval $validate_cmd
+        if [ $? == "0" ]; then
+            echo "Unable to delete $del_snapshot_name" ;
+            unblock_service ;
+            exit 1;
+        fi
+    }
+
+    if [ $# -ne 2 ] && [ "$2" != "restore_service" ]; then
+        usage 
+    fi 
+    
+    cat patch.json > tmp.json 
+    pv_name=$1
+    number_of_snapshot=$2
+    pvc_namespace=$(kubectl get pv $pv_name -o jsonpath='{.spec.claimRef.namespace}')
+    ctrl_pod_name=$(kubectl get pod -n $pvc_namespace -l openebs.io/persistent-volume=$pv_name,openebs.io/controller=jiva-controller -o jsonpath='{.items[0].metadata.name}')
+    ctrl_service_name=$(kubectl get service -n $pvc_namespace -l openebs.io/controller-service=jiva-controller-svc -o jsonpath='{.items[0].metadata.name}')
```

Comment:
this command also should include the pv-name in the filter - to handle case of multiple jiva volumes being present in the same namespace.